iati/home/tests/test_middleware.py

```diff
@@ -0,0 +1,49 @@
+"""A module of unit tests for redirects."""
+import pytest
+from django.conf import settings
+from factories import DocumentFactory
+
+
+@pytest.mark.django_db
+class TestRedirectMiddleware():
+    """Tests for the custom redirect middleware."""
+
+    def create_document(self):
+        """Call the DocumentFactory to save a new document."""
+        return DocumentFactory.create()
+
+    @pytest.mark.parametrize('redirect_mapping', [
+        ('/HoMe/', '/home/'),
+        ('/HOME/', '/home/'),
+        ('/home/', '/en/home/'),
+        ('/104/MiXeDCaSe/', 'http://reference.iatistandard.org/104/MiXeDCaSe/'),
+        ('/105/MiXeDCaSe/', 'http://reference.iatistandard.org/105/MiXeDCaSe/'),
+        ('/201/MiXeDCaSe/', 'http://reference.iatistandard.org/201/MiXeDCaSe/'),
+        ('/202/MiXeDCaSe/', 'http://reference.iatistandard.org/202/MiXeDCaSe/'),
+        ('/203/MiXeDCaSe/', 'http://reference.iatistandard.org/203/MiXeDCaSe/'),
+        ('/activity-standard/MiXeDCaSe/', 'http://reference.iatistandard.org/activity-standard/MiXeDCaSe/'),
+        ('/codelists/MiXeDCaSe/', 'http://reference.iatistandard.org/codelists/MiXeDCaSe/'),
+        ('/developer/MiXeDCaSe/', 'http://reference.iatistandard.org/developer/MiXeDCaSe/'),
+        ('/introduction/MiXeDCaSe/', 'http://reference.iatistandard.org/introduction/MiXeDCaSe/'),
+        ('/namespaces-extensions/MiXeDCaSe/', 'http://reference.iatistandard.org/namespaces-extensions/MiXeDCaSe/'),
+        ('/organisation-identifiers/MiXeDCaSe/', 'http://reference.iatistandard.org/organisation-identifiers/MiXeDCaSe/'),
+        ('/organisation-standard/MiXeDCaSe/', 'http://reference.iatistandard.org/organisation-standard/MiXeDCaSe/'),
+        ('/reference/MiXeDCaSe/', 'http://reference.iatistandard.org/reference/MiXeDCaSe/'),
+        ('/rulesets/MiXeDCaSe/', 'http://reference.iatistandard.org/rulesets/MiXeDCaSe/'),
+        ('/schema/MiXeDCaSe/', 'http://reference.iatistandard.org/schema/MiXeDCaSe/'),
+        ('/upgrades/MiXeDCaSe/', 'http://reference.iatistandard.org/upgrades/MiXeDCaSe/'),
+        (settings.STATIC_URL + "HOME/css/", "/en" + settings.STATIC_URL + "HOME/css/")
+    ])
+    def test_redirect_middleware_default(self, client, redirect_mapping):
+        """Test default behavior for redirects."""
+        response = client.get(redirect_mapping[0])
```

Comment:
To explain why we should use real URLs:

A lot of logic power can be stripped from here. `client.get()` allows a `follow` argument, meaning that the redirect will be followed - no need for two requests here. However, these URLs are returning 404s.

Not as many redirect maps need to be used for the external redirects, as they are redirected by the same url config, which may make you feel better! I recall previously that `AidType` was the test failure to pick up on this?

Therefore, this function would look something like:

```
response = client.get(redirect_mapping[0], follow=True)
assert response.url == redirect_mapping[1]
```

check that url is a response attr/property, though :)